src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.MultiAgent.cs

```diff
@@ -849,7 +849,14 @@ private void FinishRequest(StrongToWeakReference<EasyRequest> easyWrapper, CURLc
                     // ignore it and treat such failures as successes, to match the Windows behavior.
                     if (messageResult != CURLcode.CURLE_UNSUPPORTED_PROTOCOL)
                     {
-                        ThrowIfCURLEError(messageResult);
+                        // libcurl will return CURLE_RECV_ERROR (56) if proxy authentication failed when connecting to a https server,
+                        // whereas it returns CURLE_OK for a http server proxy authentication failure. We ignore this curl behavior error,
+                        // and let the user rely on response message status code to match the Windows behavior.
+                        if (messageResult != CURLcode.CURLE_RECV_ERROR ||
```

Comment:
Is this going to "cover up" any errors where we'd actually want to throw an exception for this error code?